tests/backup/backup_rbac_test.go

```diff
@@ -1255,3 +1255,291 @@ var _ = Describe("{VerifyRBACForAppAdmin}", func() {
+			_, err = DeleteBackup(scheduledBackupName, backupUid, orgID, nonAdminCtx)
+			log.FailOnError(err, "Failed to delete the backup %s of the user %s", scheduledBackupName, appUser)
+			err = DeleteBackupAndWait(scheduledBackupName, nonAdminCtx)
+			log.FailOnError(err, fmt.Sprintf("waiting for backup [%s] deletion", scheduledBackupName))
+			log.InfoD("Deleting the restore")
+			err = DeleteRestore(restoreName, orgID, nonAdminCtx)
+			dash.VerifySafely(err, nil, fmt.Sprintf("Deleting restore [%s]", restoreName))
+		})
+
+		Step(fmt.Sprintf("Validate deleting of backup schedule for the App-User [%s]", appUser), func() {
+			log.InfoD(fmt.Sprintf("Validate deleting of backup schedule for the App-User [%s]", appUser))
+			nonAdminCtx, err := backup.GetNonAdminCtx(appUser, commonPassword)
+			log.FailOnError(err, "failed to fetch user %s ctx", appUser)
+			err = DeleteSchedule(userScheduleName, SourceClusterName, orgID, nonAdminCtx)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Deleting Backup Schedule [%s] for user [%s]", userScheduleName, appUser))
+		})
+
+		Step(fmt.Sprintf("Validate deleting of source and destination cluster for the user [%s]", appUser), func() {
+			log.InfoD(fmt.Sprintf("Validate deleting of source and destination cluster for the user [%s]", appUser))
+			nonAdminCtx, err := backup.GetNonAdminCtx(appUser, commonPassword)
+			log.FailOnError(err, "failed to fetch user %s ctx", appUser)
+			for _, clusterName := range []string{SourceClusterName, destinationClusterName} {
+				err := DeleteCluster(clusterName, orgID, nonAdminCtx, false)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying deletion of cluster [%s] of the user %s", clusterName, appUser))
+				err = Inst().Backup.WaitForClusterDeletion(nonAdminCtx, clusterName, orgID, clusterDeleteTimeout, clusterCreationRetryTime)
+				log.FailOnError(err, fmt.Sprintf("waiting for cluster [%s] deletion", clusterName))
+			}
+		})
+	})
+
+	JustAfterEach(func() {
+		defer EndPxBackupTorpedoTest(scheduledAppContexts)
+		ctx, err := backup.GetAdminCtxFromSecret()
+		log.FailOnError(err, "Fetching px-central-admin ctx")
+		log.InfoD("Deleting the deployed apps after the testcase")
+		opts := make(map[string]bool)
+		opts[SkipClusterScopedObjects] = true
+		DestroyApps(scheduledAppContexts, opts)
+		log.InfoD("Deleting the px-backup objects")
+		CleanupCloudSettingsAndClusters(backupLocationMap, cloudCredName, cloudCredUID, ctx)
+		UserRules, _ := Inst().Backup.GetAllRules(ctx, orgID)
+		for _, ruleName := range UserRules {
+			err := DeleteRule(ruleName, orgID, ctx)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying deletion of rule [%s] for the Px-Admin user", ruleName))
+		}
+		log.InfoD("Switching context to destination cluster for clean up")
+		err = SetDestinationKubeConfig()
+		log.FailOnError(err, "Unable to switch context to destination cluster [%s]", destinationClusterName)
+		DestroyApps(scheduledAppContexts, opts)
```

Comment:
Not necessary I feel.

Reply:
This will be used to destroy the apps created over the destination cluster.